plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -88,16 +89,18 @@ public CachingJdbcClient(
             IdentityCacheMapping identityMapping,
             BaseJdbcConfig config)
     {
-        this(delegate, sessionPropertiesProviders, identityMapping, config.getMetadataCacheTtl(), config.isCacheMissing());
+        this(delegate, sessionPropertiesProviders, identityMapping, config.getMetadataCacheTtl(), config.isCacheMissing(), config.getCacheMaximumSize());
     }
 
     public CachingJdbcClient(
             JdbcClient delegate,
             Set<SessionPropertiesProvider> sessionPropertiesProviders,
             IdentityCacheMapping identityMapping,
             Duration metadataCachingTtl,
-            boolean cacheMissing)
+            boolean cacheMissing,
+            long cacheMaximumSize)
     {
+        validateCacheConfiguration(metadataCachingTtl, cacheMaximumSize);
```

Comment:
we could validate this configuration in a `@PostConstruct` method in a config class itself

Reply:
yeah, I think the caching client isn't guaranteed to get instantiated at server startup so with current impl the user will see runtime validation failure instead of startup I think. Same as https://github.com/trinodb/trino/pull/8652#discussion_r677234792

Reply:
Yes, that's a safer bet